src/rgw/rgw_rest_sts.cc

```diff
@@ -172,23 +172,25 @@ void RGWSTSAssumeRole::execute()
 
 RGWOp *RGWHandler_REST_STS::op_post()
 {
-  char buf[256];
-  recv_body(s, buf, s->content_length);
-  ldout(s->cct, 0) << "Content of POST: " << buf << dendl;
-  string post_body = buf;
-
-  if (post_body.find("Action") != string::npos) {
-    boost::char_separator<char> sep("&");
-    boost::tokenizer<boost::char_separator<char>> tokens(post_body, sep);
-    for (const auto& t : tokens) {
-      auto pos = t.find("=");
-      if (pos != string::npos) {
-         std::string key = t.substr(0, pos);
-         std::string value = t.substr(pos + 1, t.size() - 1);
-         ldout(s->cct, 0) << "Key: " << key << "Value: " << value << dendl;
-         s->info.args.append(key, value);
+  const auto max_size = s->cct->_conf->rgw_max_put_param_size;
+  auto [ret, data] = rgw_rest_read_all_input(s, max_size, false);
+  if (data.length() > 0) {
+    string post_body = data.to_str();
+    ldout(s->cct, 0) << "Content of POST: " << post_body << dendl;
+
+    if (post_body.find("Action") != string::npos) {
+      boost::char_separator<char> sep("&");
+      boost::tokenizer<boost::char_separator<char>> tokens(post_body, sep);
+      for (const auto& t : tokens) {
+        auto pos = t.find("=");
+        if (pos != string::npos) {
+           std::string key = t.substr(0, pos);
+           std::string value = t.substr(pos + 1, t.size() - 1);
+           ldout(s->cct, 0) << "Key: " << key << "Value: " << value << dendl;
```

Comment:
could you please change these ldouts to 10 or 20 while you're here? i don't think people will want them in their logs unless they're trying to debug something